bitnami/external-dns/templates/deployment.yaml

```
@@ -34,11 +34,7 @@ spec:
       {{- with .Values.tolerations }}
       tolerations: {{ tpl (toYaml .) $ | nindent 8 }}
       {{- end }}
-      {{- if .Values.rbac.create }}
-      serviceAccountName: {{ template "external-dns.fullname" . }}
-      {{- else }}
-      serviceAccountName: {{ .Values.rbac.serviceAccountName | quote }}
-      {{- end }}
+      serviceAccountName: {{ template "external-dns.serviceAccountName" . }}
```

Do not remove this condition: `{{- if .Values.rbac.create }}`
```suggestion
      {{- if .Values.rbac.create }}
      serviceAccountName: {{ template "external-dns.serviceAccountName" . }}
      {{- end }}
```